ui/src/hooks/useKeyboard.ts

```diff
@@ -135,9 +139,15 @@ export default function useKeyboard() {
   // It then sends the full keyboard state to the device.
   const handleKeyPress = useCallback(
     async (key: number, press: boolean) => {
-      if (rpcDataChannel?.readyState !== "open") return;
+      if (rpcDataChannel?.readyState !== "open" && !rpcHidReady) return;
       if ((key || 0) === 0) return; // ignore zero key presses (they are bad mappings)
 
+      if (rpcHidReady) {
+        console.debug("Sending keypress event via HidRPC");
+        reportKeypressEvent(key, press);
+        return;
+      }
+
       if (keyPressReportApiAvailable) {
         // if the keyPress api is available, we can just send the key press event
         sendKeypressEvent(key, press);
```

Comment:
There is a bit of duplicate logic:

- `keyPressReportApiAvailable` isn’t useful, since it’s effectively the same as rpcHidReady.
- `sendKeypressEvent()` duplicates the preceding if statement in this function. Just replace `keyPressReportApiAvailable` with `rpcHidReady` and remove the preceding if statement.